File: 0830-Positions of Large Groups/0830-Positions of Large Groups.py
class Solution:
    def largeGroupPositions(self, S):
        """
        :type S: str
        :rtype: List[List[int]]
        """
        result = []
        n = len(S)
        i = 0
        while i < n:
            start = i
            while i < n and S[i] == S[start]:
                i += 1
            if i - start >= 3:
                result.append([start, i - 1])
        return result

File: 0801-0900/0830-Positions of Large Groups/0830-Positions of Large Groups.java
class Solution {
    public List<List<Integer>> largeGroupPositions(String S) {
        List<List<Integer>> result = new ArrayList();
        int n = S.length(), i = 0;
        while (i < n) {
            int start = i;
            while (i < n && S.charAt(i) == S.charAt(start)) {
                ++i;
            }
            if (i - start >= 3) {
                result.add(Arrays.asList(start, i - 1));
            }
        }
        return result;
    }
}
